### PR TITLE
feat: apisixconsumer translator

### PR DESCRIPTION
**Pull request by tokers**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

https://github.com/apache/apisix-ingress-controller/issues/391
___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

This Pull Request adds the translation logic for `ApisixConsumer` resource, it converts this resource to an equivalent APISIX consumer object.

This PR should be reviewed and merged after #470 and #462 are merged.
___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Comment by gxthrj** [comment]
Is this PR ready to review? It is still a draft.

**Comment by tokers** [comment]
@gxthrj Please see the issue description.

File: pkg/kube/translation/apisix_consumer.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"fmt"
+
+	configv2alpha1 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/apis/config/v2alpha1"
+	apisixv1 "github.com/apache/apisix-ingress-controller/pkg/types/apisix/v1"
+)
+
+func (t *translator) TranslateApisixConsumer(ac *configv2alpha1.ApisixConsumer) (*apisixv1.Consumer, error) {
+	// As the CRD schema ensures that only one authN can be configured,
+	// so here the order is no matter.
```
Comment (gxthrj) [inline pkg/kube/translation/apisix_consumer.go:26]: Move this comment on the head of method `TranslateApisixConsumer`.

**Comment by gxthrj** [comment]
Need to deal with some conflicting files.

File: pkg/kube/translation/apisix_consumer.go (hunk 1 above)
Comment (tokers) [reply pkg/kube/translation/apisix_consumer.go:26]: This is used to describe the order to handle this authentication. Not about the `TranslateApisixConsumer` itself.

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Comment by gxthrj** [comment]
@tokers Should merge master, for e2e test.

File: pkg/ingress/controller.go [hunk 2]
```diff
@@ -435,7 +435,7 @@ func (c *Controller) namespaceWatching(key string) (ok bool) {
 	}
 	ns, _, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
-		// Ignore resource with invalid key.
+		// Ignore resource pkg/types/apisix/v1/plugin_types.gowith invalid key.
```
Comment (gxthrj) [inline pkg/ingress/controller.go:438]:
```suggestion
		// Ignore resource pkg/types/apisix/v1/plugin_types.go with invalid key.
```

**Review by gxthrj — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Pull request merged**